src/org/benf/cfr/reader/bytecode/analysis/parse/expression/ArithmeticMonOperation.java

```diff
@@ -58,6 +62,83 @@ public Dumper dumpInner(Dumper d) {
         return d;
     }
 
+    @Override
+    public Literal getComputedLiteral(Map<LValue, Literal> display) {
+        Literal l = lhs.getComputedLiteral(display);
+        if (l == null || !(l.getValue().getValue() instanceof Number))
+            return null;
+        Number lv = ((Number) l.value.getValue());
+        switch (l.getInferredJavaType().getRawType()) {
+            case BOOLEAN:
+            case BYTE:
+            case CHAR:
+            case SHORT:
+            case INT:
+                return getComputedInt(lv.intValue());
```

Comment:
May want to create some test cases to validate if mon operations on narrower types can get messed up due to treating as integers. The solution for the bi-operations was to simply cast afterwards. If the same applies here then the solution is simple.

Reply:
Doesn't cause problems. Compared by checking against:
```java
// foo = println
foo((~(short)333));
foo((~(short)222));
foo((~(short)111));
foo((~(short)-111));
foo((~(short)-333));
System.out.println();
foo((~(byte)333));
foo((~(byte)222));
foo((~(byte)111));
foo((~(byte)-111));
foo((~(byte)-333));
System.out.println();
```
The corresponding values printed when running:
```
-334
-223
-112
110
332

-78
33
-112
110
76
```
The decompiled output:
```java
Mon.foo(-334);
Mon.foo(-223);
Mon.foo(-112);
Mon.foo(110);
Mon.foo(332);
System.out.println();
Mon.foo(-78);
Mon.foo(33);
Mon.foo(-112);
Mon.foo(110);
Mon.foo(76);
```